mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/work/impl/AbstractPojoTypeIndexingPlan.java

```diff
@@ -120,7 +120,12 @@ void updateBecauseOfContainedAssociation(Object entity, int dirtyAssociationPath
 		BitSet dirtyPaths =
 				typeContext().reindexingResolver().dirtySelfOrContainingFilter().filter( dirtyAssociationPathOrdinal );
 		if ( dirtyPaths != null ) {
-			getState( identifier ).addOrUpdate( entitySupplier, dirtyPaths, false, false );
+			S state = getState( identifier );
+			// If the current entity state is "removed" and we are trying to add/update something because of
+			// the association, then we should ignore that action, since the actual value is ... removed?!
+			if ( !EntityStatus.ABSENT.equals( state.currentStatus ) ) {
+				state.addOrUpdate( entitySupplier, dirtyPaths, false, false );
+			}
```

Comment:
Hey @yrodiere 

soooo this change feels a bit too naive ... 
but depending on the order of events in the queue we can get a delete event which will trigger `updateBecauseOfContainedAssociation` and we will add things for indexing, even though the entitiy is actually removed .... 

at first I thought about adding some alternative to `updateBecauseOfContainedAssociation`, e.g. `updateBecauseOfContainedRemoved` 🤪 ... but then maybe this ^ is actually enough?